src/Aspire.Hosting.MongoDB/MongoDBServerResource.cs

```diff
@@ -6,53 +6,48 @@ namespace Aspire.Hosting.ApplicationModel;
 /// <summary>
 /// A resource that represents a MongoDB container.
 /// </summary>
-public class MongoDBServerResource : ContainerResource, IResourceWithConnectionString
+/// <param name="name">The name of the resource.</param>
+/// /// <param name="userNameParameter">A parameter that contains the MongoDb server user name, or <see langword="null"/> to use a default value.</param>
+/// <param name="passwordParameter">A parameter that contains the MongoDb server password.</param>
+public class MongoDBServerResource(string name, ParameterResource? userNameParameter = null, ParameterResource? passwordParameter = null) : ContainerResource(name), IResourceWithConnectionString
```

Comment:
Is this also a breaking change?

Reply:
So, must I create an overload with new parameters instead of named parameters with defaults?